CourseManager/ViewModels/CoursesViewModel.swift

```diff
@@ -11,15 +11,12 @@ import FirebaseCore
 import FirebaseFirestore
 import FirebaseFirestoreSwift
 import SwiftUI
-// import CoreMIDI
 
 class CoursesViewModel: ObservableObject {
     @Published var courses = [CourseInformation]()
 
-    private var db = Firestore.firestore()
-
     func fetchData() {
-        db.collection("courses").addSnapshotListener { querySnapshot, error in
+        CourseRepository.collection.addSnapshotListener { querySnapshot, error in
```

Comment:
You should remove snapshotListener, when the viewModel is destructed. Otherwise, the Firestore will hold these listeners in memory

Reply:
Hi Yura, does this work?

**I have the following code below in my viewModel:**
 
var firestoreListener: ListenerRegistration? 

 func fetchData() {
        firestoreListener = CourseRepository.collection.addSnapshotListener { querySnapshot, error in
            if let error = error {
                print("Error getting documents: \(error)")
            } else {
                guard let documents = querySnapshot?.documents else {
                    print("no documents")
                    return
                }
                self.courses = documents.compactMap { queryDocumentSnapshot -> CourseInformation? in
                    try? queryDocumentSnapshot.data(as: CourseInformation.self)
                    
                }
            }
        }
        
    }

**I have this in my View:** 
 .onDisappear {
                    self.viewModel.firestoreListener?.remove()
                }